src/alignment/mod.rs

```diff
@@ -99,36 +103,94 @@ impl Alignment {
             AlignmentOperation::Subst => cigar.push_str(&format!("{}{}", k, "X")),
             AlignmentOperation::Del => cigar.push_str(&format!("{}{}", k, "D")),
             AlignmentOperation::Ins => cigar.push_str(&format!("{}{}", k, "I")),
-            AlignmentOperation::Xclip(len) if len > 0 => {
-                cigar.push_str(&format!("{}{}", len, clip_str))
-            }
             _ => {}
         };
 
-        let mut cigar = String::new();
+        let mut cigar = "".to_owned();
+        if self.operations.is_empty() {
+            return cigar;
+        }
 
-        if !self.operations.is_empty() {
-            let mut last = self.operations[0];
-            if self.xstart > 0 {
-                cigar.push_str(&format!("{}{}", self.xstart, clip_str))
-            }
-            let mut k = 1;
-            for &op in self.operations[1..].iter() {
-                if op == last {
-                    k += 1;
-                } else {
-                    add_op(last, k, &mut cigar);
-                    k = 1;
-                }
-                last = op;
+        let mut last = self.operations[0];
+        if self.xstart > 0 {
+            cigar.push_str(&format!("{}{}", self.xstart, clip_str))
+        }
+        let mut k = 1;
+        for &op in self.operations[1..].iter() {
+            if op == last {
+                k += 1;
+            } else {
+                add_op(last, k, &mut cigar);
+                k = 1;
             }
-            add_op(last, k, &mut cigar);
-            if self.xlen > self.xend {
-                cigar.push_str(&format!("{}{}", self.xlen - self.xend, clip_str))
+            last = op;
+        }
+        add_op(last, k, &mut cigar);
+        if self.xlen > self.xend {
+            cigar.push_str(&format!("{}{}", self.xlen - self.xend, clip_str))
+        }
+
+        cigar
+    }
+
+    /// Calculate the bam cigar from the alignment struct. x is the target string
+    /// and y is the reference.
+    pub fn bam_cigar(&self, hard_clip: bool) -> Vec<bam::record::Cigar> {
```

Comment:
I would prefer to have an `impl From<Alignment> for bam::record::Cigar` in rust-htslib.

Reply:
agreed - moving these methods to rust-htslib